src/main/kotlin/app/batch/HBaseWriter.kt

```diff
@@ -28,41 +29,41 @@ class HBaseWriter(private val connection: Connection) : ItemWriter<DecompressedS
     override fun write(items: MutableList<out DecompressedStream>) {
         items.forEach {
             val fileName = it.fileName
-            val dataKeyResult: DataKeyResult = getDataKey(fileName)
-            val reader = BufferedReader(InputStreamReader(it.inputStream))
-            var line: String? = null
-            var id: String?
-            while ({ line = reader.readLine(); line }() != null) {
-                try {
-                    val parser: Parser = Parser.default()
-                    val stringBuilder = StringBuilder(line)
-                    val json = parser.parse(stringBuilder) as JsonObject
-                    id = getId(json, fileName)?.toJsonString()
-                    logger.info("Parsing DB object of id $id  in the file $fileName")
+            val filenamePattern = """(?<database>[a-z-]+)\.(?<collection>[a-z-]+)\.\d+\.json\.gz\.enc$"""
+            val filenameRegex = Regex(filenamePattern, RegexOption.IGNORE_CASE)
+            val matchResult = filenameRegex.find(fileName)
+            if (matchResult != null) {
+                val groups = matchResult.groups
+                val database = groups[1]!!.value // can assert nun-null as it matched on the regex
+                val collection = groups[2]!!.value // ditto
+                val dataKeyResult: DataKeyResult = getDataKey(fileName)
+                val reader = BufferedReader(InputStreamReader(it.inputStream))
+                var line: String? = null
+                while ({ line = reader.readLine(); line }() != null) {
+                    try {
+                        val parser: Parser = Parser.default()
+                        val stringBuilder = StringBuilder(line)
+                        val json = parser.parse(stringBuilder) as JsonObject
+                        val id = json.obj("_id")?.toJsonString()
+                        if (StringUtils.isNotBlank(id)) {
+                            val encryptionResult = encryptDbObject(dataKeyResult, line!!, fileName, id)
+                            logger.info("result: '$encryptionResult', fileName: '$fileName'.")
```

Comment:
Maybe we should not log the whole object? just a "success: $file" sort of line?

Reply:
maybe with the line number?

Reply:
Done.